dist/joii.js

```diff
@@ -922,18 +1050,223 @@ JOII.CreateProperty = function(obj, name, val, writable) {
+        proto_meta.overloads = [];
+    }
+
+    if (proto_meta.visibility !== meta.visibility) {
+        throw 'Member ' + meta.name + ': inconsistent visibility.';
+    }
+
+    proto_meta.is_abstract = false;
+
+    for (var i = 0; i < meta.parameters.length - 1; i++) {
+        if (meta.parameters[i] == '...') {
+            throw 'Member ' + meta.name + ': Variadic parameter (...) must be the last in the function parameter list.';
+        }
+    }
+
+    for (var idx = 0; idx < proto_meta.overloads.length; idx++) {
+        var functionParametersMeta = proto_meta.overloads[idx];
+
+        var foundAbstractThisLoop = false;
+        if (functionParametersMeta.is_abstract) {
+            foundAbstractThisLoop = true;
+        }
+
+        if (functionParametersMeta.parameters.length === meta.parameters.length) {
+            // this signature has the same number of types as the new signature
+            // check to see if the types are the same (duplicate signature)
+            var different = false;
+
+            for (var j = 0; j < functionParametersMeta.parameters.length; j++) {
+                if (functionParametersMeta.parameters[j] != meta.parameters[j]) {
+                    different = true;
+                }
+            }
+            if (!different) {
+                if (functionParametersMeta.is_abstract) {
+                    proto_meta.overloads.splice(idx, 1); // remove the abstract version, since we're about to add a non-abstract
+                    idx--; // adjust the idx for the changed array
+                    foundAbstractThisLoop = false; // we're removing this, so don't count it for abstract check
+                } else {
+                    if (!ignoreDuplicate) {
+                        throw 'Member ' + meta.name + '(' + meta.parameters.join(', ') + ') is defined twice.';
+                    } else {
+                        return false;
+                    }
+                }
+            }
+        }
+
+        if (foundAbstractThisLoop) {
+            proto_meta.is_abstract = true;
+        }
+    }
+
+    var functionMeta = {
+        fn: fn,
+        parameters: meta.parameters,
+        is_abstract: meta.is_abstract
+    };
+
+    prototype.__joii__.metadata[meta.name].overloads.push(functionMeta);
+
+    if (functionMeta.is_abstract) {
+        prototype.__joii__.metadata[meta.name].is_abstract = true;
+    }
+    // create function shim to validate the parameters, and allow overloading
+    //if (typeof (prototype[meta.name]) !== 'function') { // this test was preventing it from overriding toString
+    prototype[meta.name] = JOII.CreateFunctionShim(meta.name, prototype.__joii__.metadata[meta.name].overloads);
+    //}
+
+    return true;
+};
+
+
+/**
+ * Creates a function detour which checks type parameters, then dispatches to the appropriate overloaded function.
+ *
+ * @param {String}   name
+ * @param {Object}   overloads
+ */
+JOII.CreateFunctionShim = function(name, overloads) {
```

Comment:
Function isn't mean to be instantiated using the `new` keyword, therefore written in camelCase.